src/test/kotlin/pt/up/fe/ni/website/backend/controller/AccountControllerTest.kt

```diff
@@ -88,33 +112,50 @@ class AccountControllerTest @Autowired constructor(
             repository.save(testAccount)
         }
 
+        private val parameters = listOf(
+            parameterWithName("id").description(
+                "ID of the account to retrieve"
+            )
+        )
```

Comment:
The `parameterWithName` method generates a `ParameterDescriptorWithType` object that can be used for both URL and query parameters. I haven't added the option to document query parameters with the extension methods yet as currently, we have no query parameters in any of our endpoints, but I can add it now if you see fit.

Regarding the body request, it is declared in the `DocumentedJSONField` object, which declares whether it a field is present in the request, response or both. At the time of documenting,  a `FieldDescriptor` object (from the restdocs-api-spec package) is generated based on the declared options.  

The next step in granularity would be to make the `optional` and `ignored` options sensitive to whether the field is documenting a request or response, using 4 options instead of only 2 (`optionalInRequest`, `ignoredInRequest`, `optionalInResponse`, `ignoredInResponse`), but the current degree of granularity is fine for now.